app/services/travel_to_work_area_and_london_borough_service.rb

```diff
@@ -1,20 +1,81 @@
 class TravelToWorkAreaAndLondonBoroughService
-  def self.add_travel_to_work_area_and_london_borough(site:)
-    url = URI("#{Settings.mapit_url}/point/4326/#{site.longitude},#{site.latitude}?type=TTW,LBO&api_key=#{Settings.mapit_api_key}")
-    response = Net::HTTP.get(url)
-    json = JSON.parse(response)
-
-    ttw_area = json.dig(json.keys.first, "name")
-
-    london_borough = if ttw_area == "London"
-                       json
-                       .dig(json.keys.second, "name")
-                       .gsub(/ Borough Council| City Council| Corporation/, "")
-                     end
-
-    site.update_column("travel_to_work_area", ttw_area)
-    site.update_column("london_borough", london_borough)
-  rescue StandardError
-    false
+  include ServicePattern
+
+  def initialize(site:)
+    @site = site
+  end
+
+  def call
+    ttw_area = get(:travel_to_work_area)
+
+    if ttw_area == "London"
+      update_london_borough
+    elsif ttw_area
+      update_ttw_area(ttw_area)
+    else
+      false
+    end
+  end
+
+private
+
+  attr_reader :site
+
+  def update_london_borough
+    london_borough = get(:london_borough)
+    return false unless london_borough
+
+    if london_borough
+      site.travel_to_work_area = "London"
+      site.london_borough = london_borough
+      site.save!(validate: false)
+    end
+  end
+
+  def update_ttw_area(ttw_area)
+    site.travel_to_work_area = ttw_area
+    site.london_borough = nil
+    site.save!(validate: false)
+  end
+
+  def get(attribute)
+    response = Net::HTTP.get_response(url_for(attribute))
+
+    if response.is_a?(Net::HTTPSuccess)
+      json = JSON.parse(response.body)
+      name_attr = json.dig(json.keys.first, "name")
+      name_attr = name_attr&.gsub(/ Borough Council| City Council| Corporation/, "") if attribute == :london_borough
+      name_attr
+    else
+      generate_sentry_error(response.code, attribute)
+      false
+    end
+  end
+
+  def generate_sentry_error(response_code, attribute)
+    e = StandardError.new(
+      "Mapit API has returned status code #{response_code} for Site id #{site.id} whilst trying to obtain #{attribute}",
+    )
+    Raven.capture(e)
+  end
```

Comment:
An exception is instantiated but was not raised?

We need to either Capture message, or raise exception?

```suggestion
  def generate_sentry_error(response_code, attribute)
    message = "Mapit API has returned status code #{response_code} for Site id #{site.id} whilst trying to obtain #{attribute}"
    Raven.capture_message(message)
  end
  

```

```suggestion
  def generate_sentry_error(response_code, attribute)
    dead_bodies = StandardError.new(
      "Mapit API has returned status code #{response_code} for Site id #{site.id} whilst trying to obtain #{attribute}",
    )
    raise dead_bodies
  end
```

Reply:
I've updated to use `capture_message`